bokeh/charts/_chartobject.py

```diff
@@ -576,42 +239,190 @@ def palette(self):
 
         return self._palette
 
-    def reset_legend(self, marker='square'):
-        """Reset legends creating the right glyphs to represent each chart series.
+    def make_segment(self, source, x0, y0, x1, y1, color, width):
+        """ Create a segment glyph and append it to the plot.renderers list.
+
+        Args:
+            source (obj): datasource object containing segment refereces.
+            x0 (str or list[float]) : values or field names of starting ``x`` coordinates
+            y0 (str or list[float]) : values or field names of starting ``y`` coordinates
+            x1 (str or list[float]) : values or field names of ending ``x`` coordinates
+            y1 (str or list[float]) : values or field names of ending ``y`` coordinates
+            color (str): the segment color
+            width (int): the segment width
+
+        Return:
+            segment: Segment instance
+        """
+        glyph = Segment(
+            x0=x0, y0=y0, x1=x1, y1=y1, line_color=color, line_width=width
+        )
+        return GlyphRenderer(data_source=source, glyph=glyph)
+
+    def make_line(self, source, x, y, color):
```

Comment:
None of these `make_X` functions uses self, the should go into a `utils` module or something. They are almost one liners, I personally think they could be done away with entirely.